### DISCUSSION TITLE
How to set Model Hyper-Parameters in run_recbole()?

### DISCUSSION THREAD

**Discussion by johnny12150** [RUCAIBox/RecBole#758] [Q&A] [answered]
If I want to change the model hyper-parameters, is the `hyper.test` my only option?

**Comment by hyp1231** [comment, marked as answer]
If you use `run_recbole`, then you can:

1. Write down your customized hyper-parameters into a yaml file, for example, named as `your/path/to.yaml`. Then
   ```python
   from recbole.quick_start import run_recbole

   run_recbole(model='BPR', dataset='ml-100k', config_file_list=['your/path/to.yaml'])
   ```
2. OR, pass a dict of your customized hyper-parameters directly into `run_recbole`, for example,
   ```python
   from recbole.quick_start import run_recbole

   my_dict = {
     'train_batch_size': 1234
   }
   run_recbole(model='BPR', dataset='ml-100k', config_dict=my_dict)
   ```

Please refer to [API Docs](https://recbole.io/docs/recbole/recbole.quick_start.quick_start.html) for more details about `run_recbole`.

Besides, `hyper.test` is designed for [Parameter Tuning](https://recbole.io/docs/user_guide/usage/parameter_tuning.html), but not for set models' hyper-parameters.

**Reply by johnny12150** [reply]
If I take ItemKNN as an example, I can set `K` just in `YAML` to tune the hyper-parameters?
![image](https://user-images.githubusercontent.com/22369387/110362452-3fd29f00-807c-11eb-932a-a1ce4dcfd17e.png)

**Reply by hyp1231** [reply]
If you just want to set a specific value of `k`, and get the corresponding result, then you can set `k` just in `yaml`.
Actually, we will load [ItemKNN.yaml](https://github.com/RUCAIBox/RecBole/blob/master/recbole/properties/model/ItemKNN.yaml) automatically to get a default value of `k`.
https://github.com/RUCAIBox/RecBole/blob/f0f2d6bbbf5764fede6d2b6e5d6635743f0a4d99/recbole/properties/model/ItemKNN.yaml#L1-L2
If you set the value of `k` in your custom `yaml` file, it will override the value in the default `yaml` file.

But if you want to follow a parameter tuning strategy, and tune `k` to get a better result automatically. Then it's essential to write down the strategy and range of parameters into `hyper.test`.
